cmd/containerd-shim-runhcs-v1/exec_hcs.go

```diff
@@ -300,7 +300,11 @@ func (he *hcsExec) Kill(ctx context.Context, signal uint32) error {
 		}
 		return nil
 	case shimExecStateExited:
-		return errors.Wrapf(errdefs.ErrNotFound, "exec: '%s' in task: '%s' not found", he.id, he.tid)
+		log.G(ctx).WithFields(logrus.Fields{
+			"eid": he.id,
+			"tid": he.tid,
+		}).Debug("shimExecStateExited")
+		return nil
```

Comment:
Doesn't this change violate [this invariant](https://github.com/microsoft/hcsshim/blob/f50f9750ce18de430c47bff1f16fa855daaf124e/cmd/containerd-shim-runhcs-v1/exec.go#L54)?
I don't know if we rely on that property anywhere though

Reply:
Gave this a second look - The test seen [here](https://github.com/microsoft/hcsshim/blob/f50f9750ce18de430c47bff1f16fa855daaf124e/cmd/containerd-shim-runhcs-v1/exec_wcow_podsandbox_test.go#L180) makes me feel that we shouldn't violate the invariant you pointed out. 
Also, the throwing of this error isn't blocking any cleanup which we would expect to happen while stopping a container. It is merely acting as a way to tell the caller to stop trying and exit if the task we want to kill is already in the exited state. Given this, I think we could leave the code as it is today. Any thoughts?

Reply:
Wait, doesnt that (and the subsequent) test explicitly check for `ErrNotFound` on the second Kill?
Also, that test uses a [`wcowPodSandboxExec`](https://github.com/microsoft/hcsshim/blob/f50f9750ce18de430c47bff1f16fa855daaf124e/cmd/containerd-shim-runhcs-v1/exec_wcow_podsandbox.go#L52), which still retains the original [`kill` functionality](https://github.com/microsoft/hcsshim/blob/f50f9750ce18de430c47bff1f16fa855daaf124e/cmd/containerd-shim-runhcs-v1/exec_wcow_podsandbox.go#L165-L166).

Maybe checking that the exec isnt exited or a guard in [`pod.KillExec`](https://github.com/microsoft/hcsshim/blob/f50f9750ce18de430c47bff1f16fa855daaf124e/cmd/containerd-shim-runhcs-v1/pod.go#L396-L398) would make more sense and encompass both types of `shimExec`s?

Reply:
Checking that the exec is not in the exited state is a good idea. Need to just ensure that we aren't relying on the ErrNotFound property elsewhere. Will verify this and also check how the Linux shim code is handling it

Reply:
I think those links may be broken

Reply:
Where is the current error being logged at? Maybe we can check that it's not ErrNotFound before logging it but still return the error. Not sure how much we care about the extraneous log that we get today though. Given this, I think it would be fine to just keep the behavior as is today.

Reply:
Sure, I'm fine with that

Reply:
Sync-ed up with Kathryn offline. Since the vendor function athttps://github.com/microsoft/hcsshim/blob/f50f9750ce18de430c47bff1f16fa855daaf124e/vendor/google.golang.org/grpc/status/status.go#L52 logs any error that get returned from shim, it might not be a good idea to filter the logging on shimExecStateExited since it is in the vendor code path. All the necessary clean up is also happening as expected along the container stop code flow. Will leave the code as is for now.